### DISCUSSION TITLE
Concurrent connexions for vscode

### DISCUSSION THREAD

**Discussion by Rizialdi** [coder/code-server#2634] [Q&A]
Good morning / evening,

I hope you are well. I recently came across this project in an attempt to imagine how websites such as [Coderpad](https://coderpad.io/) or [Interviewing](https://interviewing.io/) work behind the scene. And yes, fellow coders, I want to do the same 😊 for a niche 🤫. I watched videos on how to set code-server in your server, but still can't understand how it's possible to maintain multiple concurrent connexions to multiple code environments 🤔. I know that one way of doing this (concurrent connexions)  is the use of Elixir/Phoenix. But that's where my knowledge stops.

Could it be possible to use code-server for such a case with some other tool ⚙️? 
Also, for you already users of code-server, could the vs-code extension [live-share](https://visualstudio.microsoft.com/fr/services/live-share/) be used for remote collaboration and audio call 📞?

I hope this message is clear 🙏. Thank you in advance for your inputs 🤗.

**Comment by oxy** [comment]
Hi! Not sure what exactly you mean by concurrent connections - but on live-share, we cannot actually use Microsoft's VS Live Share as that is a proprietary, Microsoft-only extension that can only be used with Microsoft products.

**Reply by Rizialdi** [reply]
Thanks a lot for your inputs Oxy. By concurrent I mean having multiple vscode session running at the same time. And about vscode's live-share it a bit weird. I thought it was a simple extension.
Do you have any idea how a website such as Coderpad could work ?

**Reply by code-asher** [reply]
You can access code-server concurrently but there's a bug with that right now so I wouldn't recommend it.

If I had to bet these websites are launching separate VMs or containers for each code-server instance.

https://github.com/cdr/code-server/blob/v3.8.0/doc/FAQ.md#multi-tenancy

**Reply by Rizialdi** [reply]
Thank you @code-asher for your answer. At one point I thought no one had an answer. I did dig a bit and I found out that under the hood interviewing.io uses coderpad.io and that coderpad.io under the hood uses repl.it. In the end it's simpler to choose existing solutions since it doesn't cost that much and scalability is guaranteed. Thanks guys for your help.
